front-end-tools/src/Main.tsx

```diff
@@ -217,172 +64,17 @@ export default function Main(props: ConnectionProps) {
-                .then((value) => {
-                    if (value === undefined) {
-                        setIsModuleReferenceAlreadyDeployedStep2(false);
-                    } else {
-                        setIsModuleReferenceAlreadyDeployedStep2(true);
-                    }
-                })
-                .catch(() => {
-                    setIsModuleReferenceAlreadyDeployedStep2(false);
-                });
-        }
-    }, [connection, account, client, moduleReference]);
-
-    useEffect(() => {
-        if (
-            moduleReference !== '' &&
-            moduleReferenceCalculated !== '' &&
-            moduleReferenceCalculated !== moduleReference
-        ) {
-            setShouldWarnDifferenceModuleReferences(true);
-        } else {
-            setShouldWarnDifferenceModuleReferences(false);
-        }
-    }, [moduleReference, moduleReferenceCalculated]);
-
-    useEffect(() => {
-        if (inputParameterTemplate !== '' && hasInputParameter === false) {
-            setShouldWarnInputParameterInSchemaIgnored(true);
-        } else {
-            setShouldWarnInputParameterInSchemaIgnored(false);
-        }
-    }, [inputParameterTemplate, hasInputParameter]);
-
-    useEffect(() => {
-        setSchemaError('');
-        setInputParameterTemplate('');
-
-        let template = '';
-
-        if (contractName !== '' && (useModuleFromStep1 || uploadedModuleSchemaBase64 !== '')) {
-            try {
-                const inputParamterTypeSchemaBuffer = getInitContractParameterSchema(
-                    toBuffer(useModuleFromStep1 ? embeddedModuleSchemaBase64 : uploadedModuleSchemaBase64, 'base64'),
-                    contractName,
-                    2
-                );
-
-                template = displayTypeSchemaTemplate(inputParamterTypeSchemaBuffer);
-
-                setInputParameterTemplate(template);
-            } catch (e) {
-                if (useModuleFromStep1) {
-                    setSchemaError(
-                        `Could not get embedded input parameter schema from the uploaded module. \nUncheck "Use Module from Step 1" checkbox to manually upload a schema. Original error: ${e}`
-                    );
-                } else {
-                    setSchemaError(`Could not get input parameter schema from uploaded schema. Original error: ${e}`);
-                }
-            }
-        }
-
-        if (dropDown === 'array') {
-            const element = inputParameterTextAreaRef.current as unknown as HTMLSelectElement;
-            element.value = getArrayExample(template);
-        } else if (dropDown === 'object') {
-            const element = inputParameterTextAreaRef.current as unknown as HTMLSelectElement;
-            element.value = getObjectExample(template);
-        }
-    }, [hasInputParameter, useModuleFromStep1, contractName, uploadedModuleSchemaBase64, dropDown]);
-
     useEffect(() => {
         if (connection && client && account) {
```

Comment:
This bottom `useEffect` triggers when the page is loaded to immediately display the account balance. While the top `useEffect` starts after the first `REFRESH_INTERVAL` seconds have passed to then periodically poll for any updates in the account balance.

Without the bottom `useEffect`, you would have to wait until the first `REFRESH_INTERVAL` seconds have passed before seeing any account balance info.

Reply:
Yes but what I was referring to is that if the body of the interval callback and the initial load is the same, it would make sense to have make them as one function used twice